hedera-node/src/main/java/com/hedera/services/state/migration/MigrationRecordsManager.java

```diff
@@ -124,6 +143,36 @@ private TransactionBody.Builder synthStakingFundCreate(final long autoRenewPerio
 		return TransactionBody.newBuilder().setCryptoCreateAccount(txnBody);
 	}
 
+	private void publishContractFreeAutoRenewalRecords(final Instant now) {
+		final var changedContracts = new HashMap<EntityNum, Long>();
+		accounts.get().forEach((a, b) -> {
+			if (b.isSmartContract()) {
+				changedContracts.put(a, b.getExpiry());
+			}
+		});
+
+		for (var entry : changedContracts.entrySet()) {
+			final var contractNum = entry.getKey().toEntityId();
+			final var newExpiry = entry.getValue();
+
+			final var receipt = new TxnReceipt();
+			receipt.setAccountId(contractNum);
+
+			final var synthBody = syntheticTxnFactory.synthContractAutoRenew(contractNum.asNum(), newExpiry);
+			final var memo = String.format(CONTRACT_UPGRADE_MEMO, contractNum.num(), newExpiry);
+			final var txnId = new TxnId(contractNum, MISSING_INSTANT, false, USER_TRANSACTION_NONCE);
+			final var synthRecord = ExpirableTxnRecord.newBuilder()
+					.setTxnId(txnId)
```

Comment:
The `RecordsHistorian` will [automatically set an appropriate `TransactionID` with `nonce`](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/records/TxnAwareRecordsHistorian.java#L213) when streaming your record, so this isn't necessary.

Reply:
Got it ! Fixed ! Thanks !